kyuubi-common/src/main/scala/org/apache/kyuubi/Logging.scala

```diff
@@ -88,13 +88,55 @@ trait Logging {
     if (!Logging.initialized) {
       Logging.initLock.synchronized {
         if (!Logging.initialized) {
-          initializeLogging(isInterpreter)
+          Logging.initializeLogging(isInterpreter, loggerName, () => logger)
         }
       }
     }
   }
+}
+
+object Logging {
+  @volatile private var useDefault = false
+  @volatile private var defaultRootLevel: String = _
+  @volatile private var initialized = false
+  val initLock = new Object()
+
+  private[kyuubi] def isLog4j12: Boolean = {
+    // This distinguishes the log4j 1.2 binding, currently
+    // org.slf4j.impl.Log4jLoggerFactory, from the log4j 2.0 binding, currently
+    // org.apache.logging.slf4j.Log4jLoggerFactory
+    val binderClass = StaticLoggerBinder.getSingleton.getLoggerFactoryClassStr
+    "org.slf4j.impl.Log4jLoggerFactory".equals(binderClass)
+  }
+
+  private[kyuubi] def isLog4j2: Boolean = {
+    // This distinguishes the log4j 1.2 binding, currently
+    // org.slf4j.impl.Log4jLoggerFactory, from the log4j 2.0 binding, currently
+    // org.apache.logging.slf4j.Log4jLoggerFactory
+    val binderClass = StaticLoggerBinder.getSingleton.getLoggerFactoryClassStr
+    "org.apache.logging.slf4j.Log4jLoggerFactory".equals(binderClass)
+  }
+
+  /**
+   * Return true if log4j2 is initialized by default configuration which has one
+   * appender with error level. See `org.apache.logging.log4j.core.config.DefaultConfiguration`.
+   */
+  private def isLog4j2DefaultConfigured(): Boolean = {
+    val rootLogger = LogManager.getRootLogger.asInstanceOf[Log4jLogger]
+    // If Log4j 2 is used but is initialized by default configuration,
+    // load a default properties file
+    // (see org.apache.logging.log4j.core.config.DefaultConfiguration)
+    rootLogger.getAppenders.isEmpty ||
+    (rootLogger.getAppenders.size() == 1 &&
+      rootLogger.getLevel == Level.ERROR &&
+      LogManager.getContext.asInstanceOf[LoggerContext]
+        .getConfiguration.isInstanceOf[DefaultConfiguration])
+  }
 
-  private def initializeLogging(isInterpreter: Boolean): Unit = {
+  private def initializeLogging(
+      isInterpreter: Boolean,
+      loggerName: String,
+      logger: () => Logger): Unit = {
```

Comment:
thuck could be used here to simplify code
```suggestion
      logger: => Logger): Unit = {
```